src/mono/mono/utils/mono-dl.c

```diff
@@ -469,7 +469,7 @@ dl_build_path (const char *directory, const char *name, void **iter, dl_library_
 		return NULL;
 
 	if (need_prefix) {
-		prlen = strlen (mono_dl_get_so_prefix ());
+		prlen = (int)strlen (mono_dl_get_so_prefix ());
```

Comment:
This seems like a good example where `prlen` as `size_t` is the right thing and it should just work. No?